tests/Benchmarks/Pipelines/E2EPipeline.cs

```diff
@@ -0,0 +1,69 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Text;
+using System.IO.Pipelines;
+using System.IO.Pipelines.Samples;
+using System.Text;
+using System.Text.Formatting;
+using System.Text.Json;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Attributes.Jobs;
+using BenchmarkDotNet.Engines;
+
+namespace Benchmarks.Pipelines
+{
+    [SimpleJob(RunStrategy.Monitoring)] // the setting for benchmarks with not-steady state
```

Comment:
Why is this necessary here? We previously ran it multiple times and reported the result as part of xunit.perf.

Reply:
Any thoughts on this, @adamsitnik?

Reply:
@ahsonkhan The default settings of BenchmarkDotNet use some heuristics to determine how many times to run the benchmark to get stable results. This is perfect for steady-state benchmarks. 

However, benchmarks which don't have a steady state (like heavy IO benchmarks) should not use the default settings. Their distribution is typically very different, which does not work well with our heuristics.

[Monitoring](https://benchmarkdotnet.org/Guides/ChoosingRunStrategy.htm) strategy runs it a certain amount of times, does not pre-warmup the code and does not exclude the outliers. It's preferred strategy for benchmarks like this one.